parts/k8s/windowscontainerdfunc.ps1

```diff
@@ -161,12 +162,18 @@ function Install-Containerd {
   $formatedconf = $(($CNIConfDir).Replace("\", "/"))
   $sandboxIsolation = 0
   $windowsReleaseId = (Get-ItemProperty "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion").ReleaseId
+  $windowsCurrentBuild = (Get-ItemProperty "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion").CurrentBuild

```

Comment:
should we switch all versions to use the `CurrentBuild` id? Does this return the same value as `ReleaseId` for all previous versions of Windows server (2019/2004/etc)?

Reply:
The code uses `CurrentBuild` for all cases unless it can't. In that case, it relies on `ReleaseId` as a fallback

Reply:
looks like it `ReleaseId` returns `1809` and `CurrentBuild` return `17763` for WS2019:

```
PS C:\Users\capi> (Get-ItemProperty "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion").CurrentBuild 
17763 
PS C:\Users\capi> (Get-ItemProperty "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion").ReleaseId 
1809 
```

Reply:
Yes. I use Select-Windows-Version as a map from `CurrentBuild` to the value `ReleaseId` would return
```
switch ($buildNumber) {
    "17763" { return "1809" }
    "18362" { return "1903" }
    "18363" { return "1909" }
    "19041" { return "2004" }
    "19042" { return "20H2" }
    "20348" { return "ltsc2022" }
    Default { return "" } 
}
```
In your example, 17763 => 1809

Reply:
logic looks good here. Let's wrap this into a single function that just returns windows version to hide this complexity from the `Install-Containerd` function